.github/workflows/buf.yml

```diff
@@ -0,0 +1,20 @@
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  steps:
+    - uses: actions/checkout@v2
+    - uses: bufbuild/buf-setup-action@v0.5.0
+    - uses: bufbuild/buf-lint-action@v1
+      with:
+        input: 'prompb'
+    - uses: bufbuild/buf-breaking-action@v1
+      with:
+        input: 'prompb'
+        against: 'https://github.com/prometheus/prometheus.git#branch=main,ref=HEAD~1,subdir=prompb'
```

Comment:
I don't think you need the `ref=HEAD~1` here - if you're comparing against the GitHub repository directly, the `main` branch in the `prompb` sub-directory should be sufficient, no?

Reply:
Ah after revisiting [the documentation](https://docs.buf.build/ci-cd/github-actions#buf-push), I may have misspoke. This looks fine as-is.

Reply:
I've used the example in https://github.com/bufbuild/buf-push-action/blob/main/README.md to stay as close as possible to the official documentation. Buf if the main branch is enough let's simplify it :)

Reply:
no problem I've revert it :)